### ISSUE TITLE
Question of the performance of model faster_rcnn_nas 

### ISSUE THREAD

**Issue by chenyuZha** [tensorflow/models#2905]
### System information
- **What is the top-level directory of the model you are using**: Object Detection
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow version (use command below)**: 1.4
- **CUDA/cuDNN version**: cuda 8.0
- **GPU model and memory**: GeForce GTX 1080 Ti x2

### Describe the problem
I recently downloaded the model [faster_rcnn_nas](http://download.tensorflow.org/models/object_detection/faster_rcnn_nas_coco_2017_11_08.tar.gz) from the model zoo. For the COCO data set, this model achieve a MAP=43 , which is superior than model **faster_rcnn_inception_resnet_v2_atrous_coco**(MAP=37). When I trained with my own data set, the performance of faster_rcnn_nas is far away from faster_rcnn_inception_resnet_v2_atrous: for about 20k iterations, its LOSS is about 2 and an other model is below 1. When I do the inference to check the results, the faster_rcnn_nas was very bad to detect the objects in validation images.

I take the configuration file from **sample/config**, I didn't change anything besides I set 'dropout=True'.
I would like to see if anyone has the same issue as mine , or it is just the problem of my own data set ? Or this could be that  **the pre-trained model is not training enough steps before it is released**?

**Comment by tatatodd** [comment]
@jch1 might have some ideas here.

**Comment by shlens** [comment]
I am sorry that you have had such trouble with the model but thank you for the clear documentation of the problem. I have a few comments and questions that might require some investigation on your part:
1. What is the distribution of image resolutions in your dataset? The exported model runs on high resolution (1200x1200) images and if the average image resolution in your dataset is less then this, the model might perform selectively worse on your problem.
2. What is the distribution of aspect ratios? The exported model config reshapes all images to be squares. There is a [TODO](https://github.com/tensorflow/models/blob/master/research/object_detection/models/faster_rcnn_nas_feature_extractor.py#L111) comment (and [here](https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/faster_rcnn_nas_coco.config#L11)) in the model to instead support a resizing method that maintains the aspect ratio of the image. If your images are far from square, this may lead to poor performance.
3. How many first_stage_max_proposals are you using in your configuration? The number in the exported model was set [artificially low](https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/faster_rcnn_nas_coco.config#L46) (50) in order to keep inference time reasonable. In the original [NAS paper](https://arxiv.org/abs/1707.07012), we actually employ 500 proposals to achieve the best performance. I would change this value accordingly and retrain.

**Comment by chenyuZha** [comment]
@shlens 
1.The resolutions of my images is mostly below 600, because I've resized 3 times smaller to have more images to train in my tfRecord. (which is 9G)
2. I've tried to set the keep_aspect_ratio_resizer {
        min_dimension: 600
        max_dimension: 1024
      } in the configuration file of **faster_rcnn_nas**, as you said above, this is not available yet , so I set fixed_shape_resizer {
        height: 800
        width: 800
      } instead. 
3. I've modified  the first_stage_max_proposals=300 instead of 50. I'll change it to 500 to see if I got a better performance.

**Comment by shlens** [comment]
If your images are below 600 x 600 in resolution, then artificially resizing the images to 800 x 800 (or higher) resolutions will only waste computation and potentially slow down the learning speed of the network. Employing a keep_aspect_ratio_resizer with a min_dimension of 600 would therefore be appropriate. Since that resizing operation is not available for NAS, the next best option is to used a fixed height and width of 600.

That said, the [faster_rcnn_inception_resnet_v2_atrous_coco.config](https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/faster_rcnn_inception_resnet_v2_atrous_coco.config#L13) was trained for images with a min_dimension of 600 on one side of the image. In contrast, [faster_rcnn_nas_coco](https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/faster_rcnn_nas_coco.config#L16) is trained at a resolution of 1200 x 1200. My best guess is that so long as the images being employed are at an effective resolution of 600 pixels, then the best solution is to train faster_rcnn_nas_coco from scratch at a lower resolution (600 x 600). In lieu of that, so long as the images are low resolution, then faster_rcnn_inception_resnet_v2_atrous_coco may provide the best performance.

**Comment by shlens** [comment]
Closing due to inactivity.
